readthedocs/projects/tasks.py

```diff
@@ -1564,6 +1566,12 @@ def _create_imported_files(*, version, commit, build, search_ranking):
                     page_rank = rank
                     break
 
+            ignore = False
+            for pattern in search_ignore:
+                if fnmatch(relpath, pattern):
```

Comment:
We are doing this iteration and comparison for `search_ranking` above and here for `search_ignore`. Do we know the performance of this when having many patterns?

I don't think it's too important to worry about right now, but if we already have the list of all the files imported, we could use `fnmatch.filter` https://docs.python.org/3/library/fnmatch.html#fnmatch.filter that it's already optimized.

Reply:
That function allows you to match several files with one pattern, here we are matching several rules against several files, and we are iterating file per file to create it with the right values already.